R/java_unpack.R

```diff
@@ -0,0 +1,79 @@
+#' Unpack a Java distribution file into cache directory
+#' 
+#' @inheritParams java_install
+#' @inheritParams global_quiet_param
+#' @return A `character` vector containing of length 1 containing the path to the unpacked Java directory.
+#' @keywords internal
+java_unpack <- function(
```

Comment:
Pulling this out into a separate function makes a lot of sense to me 😄

Reply:
To me it became obvious once I started moving things around for `use_java()`